### PR TITLE
Fix slow run filter

### PR DESCRIPTION
**Pull request by meksor**
The performance improvement from [PR#56](https://github.com/iiasa/ixmp4/pull/56) didnt make it into the run filter. 
This PR intends to fix that and fix some behaviour of the iamc filter as well..

### CODE REVIEW

File: ixmp4/data/db/run/filter.py [hunk 1]
```diff
@@ -13,15 +13,27 @@ class IamcRunFilter(filters.BaseFilter, metaclass=filters.FilterMeta):
...
 class RunFilter(base.RunFilter, metaclass=filters.FilterMeta):
-    iamc: IamcRunFilter | filters.Boolean
+    iamc: IamcRunFilter | filters.Boolean | None = None
+
+    def join_datapoints(self, exc: db.sql.Select, session=None):
+        if not utils.is_joined(exc, TimeSeries):
+            exc = exc.join(TimeSeries, onclause=TimeSeries.run__id == Run.id)
+        return exc
+
+    def filter_iamc(self, exc, c, v, session=None):
```
Comment (glatterf42) [inline ixmp4/data/db/run/filter.py:27]: I don't see `c` being used anywhere in the function, is that correct? 
Why don't we typehint `exc, c, v` to make more explicit what we expect here?

**Review by glatterf42 — COMMENTED** [review]
Looks good to me in principle, just one comment.

File: ixmp4/data/db/run/filter.py (hunk 1 above)
Comment (meksor) [reply ixmp4/data/db/run/filter.py:27]: Happy to review or write a cleanup PR.
These functions are never type hinted, but i would adjust them /all/ while we are at it, not just this one.

**Review by danielhuppmann — APPROVED** [review]
Thanks @meksor!

**Pull request merged**